### PR TITLE
Userpermission

### PR DESCRIPTION
**Pull request by kate-walters**
Pls leave feedback :)

### CODE REVIEW

File: app/views/notes/_form.html.erb [hunk 1]
```diff
@@ -23,6 +23,24 @@
     </div>
 
     <div>
+<<<<<<< HEAD
```
Comment (MattRice12) [inline app/views/notes/_form.html.erb]: You have a few of these conflict labels laying around

File: app/views/notes/_form.html.erb [hunk 2]
```diff
@@ -23,6 +23,24 @@
...
+      <%= f.select :user_id, options_from_collection_for_select(User.all, :id, :name) %>
+    </div>
+
+    <div>
+      <label for="permission_type">Permission Type: </label>
+      <%= f.select :permission_id, options_from_collection_for_select(Permission.all, :permission_id, :permission_type) %>
+    </div>
+
+    <div>
+=======
```
Comment (MattRice12) [inline app/views/notes/_form.html.erb]: and these at the bottom too

File: app/controllers/notes_controller.rb [hunk 3]
```diff
@@ -32,30 +42,44 @@ def edit
...
     else
-      render html: 'Note not found', status: 404
+      flash[:alert] = "You do not have permission to view this page."
+      redirect_to root_path
     end
   end
 
   def destroy
+    if has_permission?
     note = Note.find(params[:id])
```
Comment (MattRice12) [inline app/controllers/notes_controller.rb]: `note = Note.find(params[:id])` should go before the if statement

**Comment by MattRice12** [comment]
👍 Looks good, just left some notes on cleaning a few lines up.

**Comment by rluck0419** [comment]
looks good to me, after reading matt's comments too. this will require me to change a bit of the user traversal story - but you put the logic in the controller instead of the view, which should be easier/better practice than how I originally envisioned it 👍

File: app/controllers/photos_controller.rb [hunk 4]
```diff
@@ -60,4 +75,8 @@ def destroy
   def photo_params
     params.require(:photo).permit(:user_id, :category_id, :name, :desc, :upload)
   end
+
+  def has_permission?(photos)
+    Photo.user_id == current_user.id || current_user.admin?
+  end
```
Comment (DaneCarmichael) [inline app/controllers/photos_controller.rb]: The argument coming in is photos. Photos is not used in the method call. Also Photo.user_id, what photo is that?

File: app/controllers/notes_controller.rb [hunk 5]
```diff
@@ -32,30 +42,44 @@ def edit
...
+      redirect_to root_path
     end
   end
 
   private
   def note_params
     params.require(:note).permit(:user_id, :category_id, :name, :body, :file_type)
   end
+
+  def has_permission?(notes)
```
Comment (DaneCarmichael) [inline app/controllers/notes_controller.rb]: The argument coming in is notes. Notes is not used in the method call. Also Note.user_id, what photo is that?

**Comment by DaneCarmichael** [comment]
In addition to the notes I just made, Matt and Rob's concerns need to be addressed as well.

**Pull request closed without merging**